File: 0057-insert-interval/0057-insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        answer = [] 
        index = 0 
        while index < len(intervals):
            lst = intervals[index] 
            if lst[1] < newInterval[0]:
                answer.append(lst) 
            else:
                break 
            
            index += 1
        

        while index < len(intervals):
            currMin = intervals[index][0] 
            currMax = intervals[index][1] 

            if currMin <= newInterval[1]:
                newInterval[0] = min(newInterval[0], currMin) 
                newInterval[1] = max(newInterval[1], currMax)
            else:
                break

            index += 1 
        
        answer.append(newInterval) 

        while index < len(intervals):
            answer.append(intervals[index])
            index += 1

        
        return answer

File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        if(intervals.length == 0) return new int[][]{newInterval};   // 2차원 배열로 감싸기
        List<int[]> answer = new ArrayList<>(); 
        int end = 0, n = intervals.length; 
        while(end < n && intervals[end][1] < newInterval[0]){
            answer.add(intervals[end]); 
            end++; 
        }

        int currMin = newInterval[0]; 
        int currMax = newInterval[1]; 
        while(end < n && intervals[end][0] <= currMax){
            currMin = Math.min(currMin, intervals[end][0]); 
            currMax = Math.max(currMax, intervals[end][1]); 
            end++; 
        }

        answer.add(new int[]{currMin, currMax}); 

        while(end < n) answer.add(intervals[end++]); 

        return answer.toArray(new int[answer.size()][]); 
    }
}